src/Aspire.Hosting.Azure.KeyVault/AzureKeyVaultResourceExtensions.cs

```diff
@@ -63,6 +64,27 @@ public static IResourceBuilder<AzureKeyVaultResource> AddAzureKeyVault(this IDis
                 Value = keyVault.Properties.VaultUri
             });
 
+            // Process all secret annotations
+            if (infrastructure.AspireResource.TryGetAnnotationsOfType<AzureKeyVaultSecretAnnotation>(out var secretAnnotations))
+            {
+                foreach (var secretAnnotation in secretAnnotations)
+                {
+                    var paramValue = secretAnnotation.Value.AsProvisioningParameter(infrastructure, isSecure: true);
+
+                    var secret = new KeyVaultSecret(Infrastructure.NormalizeBicepIdentifier($"secret-{secretAnnotation.SecretName}"))
```

Comment:
```suggestion
                    var secret = new KeyVaultSecret(Infrastructure.NormalizeBicepIdentifier($"secret_{secretAnnotation.SecretName}"))
```

Might as well make it correct to start with.